File: backend/src/main/java/com/optiroute/backend/entity/Driver.java
package com.optiroute.backend.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "driver")
@Getter
@Setter
public class Driver extends EntityUtils {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false, unique = true)
    private String email;

    private String phoneNumber;

    @Column(nullable = false)
    private Boolean isActive = true;

    @Column(name = "monthly_salary", precision = 10, scale = 2)
    private BigDecimal monthlySalary;

    @Column(name = "monthly_working_hours", precision = 6, scale = 2)
    private BigDecimal monthlyWorkingHours;
}

File: frontend/src/models/Driver.ts
export interface Driver {
    id: number
    email: string
    firstName: string
    lastName: string
    phoneNumber: string
    isActive: boolean

    //TMP
    monthlySalary: number
    monthlyWorkingHours: number
}

export interface DriverDetails extends Driver {
    createdAt: string
    updatedAt: string
}

export interface CreateDriverRequest {
    email: string
    firstName: string
    lastName: string
    phoneNumber: string
    monthlySalary: number
    monthlyWorkingHours: number
}

export interface UpdateDriverRequest {
    email: string
    firstName: string
    lastName: string
    phoneNumber: string
    monthlySalary: number
    monthlyWorkingHours: number
}